pandas/tests/indexes/period/test_period_range.py

```diff
@@ -51,6 +51,11 @@ def test_construction_from_period(self):
         result = period_range(start=start, end=end, freq="Q", name="foo")
         tm.assert_index_equal(result, expected)
 
+        #test for issue #21793
+        idx = period_range(start=start, end=end, freq="Q", name="foo")
+        result = idx == idx.values
+        assert result.all()
```

Comment:
Could you construct the expected result and use `tm.assert_numpy_array_equal` instead?

```
result = idx == idx.values
expected = np.array(...)
tm.assert_numpy_array_equal(result, expected)
```